### ISSUE TITLE
ragg2 not working on Arch and MacOSX

### ISSUE THREAD

**Issue by deepakchethan** [radare/radare2#12931]
# This template is meant for bug reports, if you have a feature request, please be as descriptive as possible and delete the template

Make sure you are testing using the latest git version of radare2 before submitting any issue.

*If you would like to report a bug, please fill the template bellow*

### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Arch Linux x86_64
| File format of the file you reverse (mandatory)      | ELF 
| Architecture/bits of the file (mandatory)            | x86/32
| r2 -v full output, **not truncated** (mandatory)     | radare2 3.2.0-git 20551 @ linux-x86-64 git.3.1.3-198-gbfa346286
commit: bfa3462866670931cb2efe9ae87f57ed2c477063 build: 2019-01-10__10:19:00

### Expected behavior
ragg2 hi.c must give a tiny binary
### Actual behavior
It is giving me an error
```
Buffer truncated at 124 because of the RAsmOp abuse.
Input: eb0e662e0f1f8400000000000f1f400050bf01000000488d359f000000ba06000000e81900000031ff89442404e85e00000031d289042489d059c30f1f440000897c24fc48897424f0895424ec8b5424fc895424dc488b7424f048897424d08b5424ec895424cc8b7c24dc488b7424d08b5424ccb8010000000f05488
Segmentation fault (core dumped)
```
### Steps to reproduce the behavior 
- Please share the binary if it is shareable by drag and dropping it here in a zip archive (mandatory)
- Use asciinema to describe the issue and share the link here (mandatory if u can't share the binary) 
- Use code markdown `CODE` to make your code visible
- Or even better, create a Pull-Request for the https://github.com/radare/radare2-regressions containing the test case examples can be found in the t/ folder see for example https://github.com/radare/radare2-regressions/blob/master/t/cmd_search#L7

The source code of hi.c
```
main() {
    write(1, "Hello", 6);
    exit(0);
}
```
and used the ragg2 command as
```
ragg2 hi.c
```

### Additional Logs, screenshots, source-code,  configuration dump, ...

Drag and drop zip archives containing the Additional info here, don't use external services or link.
The zip file contains the binary and the screenshot of the error message.
[ragg.zip](https://github.com/radare/radare2/files/2804274/ragg.zip)

I think the issue is because of the assumption within `r_asm_pseudo_hex` method in `asm.c`. Which says that "Assume input fits inside op->buf". I tested and I got the op-buf size as 56 and input size as 249.

**Comment by deepakchethan** [comment]
Segmentation fault seems to go away when I increase the RStrBuf's buf from 32 to 1024.

**Comment by deepakchethan** [comment]
Okay, after debugging it for some time. I realized that the bug is due to the overflow of RAsmOp's buf. Since it is not bounded. It keeps growing and occupies the entire RStrBuf of buf, buf_asm, buf_hex. Since the ptr of buf is being overwritten, that is being passed as an accessor, i.e., used to fetch the *val value. Thus the segmentation fault. The memory content is:

<img width="1667" alt="screen shot 2019-01-30 at 10 05 21 pm" src="https://user-images.githubusercontent.com/20895544/51997446-bf283880-24dc-11e9-8252-72e1fb3e16f0.png">

The part of the code that is making this error is:

<img width="579" alt="screen shot 2019-01-30 at 10 12 15 pm" src="https://user-images.githubusercontent.com/20895544/51997492-d9faad00-24dc-11e9-9921-017dcc34fd65.png">

Coming to the main point, since the RStrBuf's buf is bounded to 32 Bytes, I wanted to know if the copying must stop after 32 bytes or should it overwrite the first 32 bytes again. Please let me know how should I proceed.

**Comment by radare** [comment]
No. Strbuf is not bounded to 32. You should use the api not access the fields inside. Thats why its crashing. This change happened in 3.0 and seems like there are some leftovers still

**Comment by deepakchethan** [comment]
In r_strbuf.h the `buf` variable is declared as buf[32]. The input is more than 200 bytes for the c program with write() method. 
Correct me if am wrong, the code in place now converts each input character to byte. There is no way to use both r_strbuf or r_str API for this. Since the computation and assignment happens as a byte level.
